### PR TITLE
Moves test code into test blocks

### PR DESCRIPTION
**Pull request by Cypher1**
This should improve some unreliable behavior when using flags (as test code was executing before the flags were set).

I think we still need to find a place to inject a `Flags.reset()` that will always run before

### CODE REVIEW

File: tools/sigh.ts [hunk 1]
```diff
@@ -723,6 +724,7 @@ function runTests(args: string[]): boolean {
     for (const test of tests) {
       chain.push(`
         import {mocha} from '${mochaInstanceFile}';
+        import {Flags} from '${flagsInstanceFile}';
```
Comment (galganif) [inline tools/sigh.ts]: why is this needed? tests that need this can import the file?I would leave this change out unless you can provide a rationale for when to add imports here.

File: src/dataflow/analysis/tests/analysis-test.ts [hunk 2]
```diff
@@ -394,11 +394,6 @@ describe('Solver', () => {
   });
 });
 
-const entityString = Flags.useNewStorageStack ?
```
Comment (galganif) [inline src/dataflow/analysis/tests/analysis-test.ts]: maybe have a function getEntityString() so you can run in the test as well as avoid the duplication. Alternatively initialise it in beforeEach

File: tools/sigh.ts (hunk 1 above)
Comment (Cypher1) [reply tools/sigh.ts]: Found out a bit more about this.
The problem is that Flags can be imported long before the `globals` test flags are set up, meaning that we don't incorporate the values into the `Flags` and get bad state (depending on test's interaction with `Flags`).

I've added a comment that we need the reset to update the flags with the command line flags.

Maybe there's a better way to do this (lazily getting the flags and command line flags at the first call) but I think it would be more complex and possibly add overhead to running and debugging any use of flags.

Is there a better way? Thanks for catching this, I hadn't thought it through enough when first using this as a fix.

File: src/dataflow/analysis/tests/analysis-test.ts (hunk 2 above)
Comment (Cypher1) [reply src/dataflow/analysis/tests/analysis-test.ts]: Updated, thanks
